presto-hive/src/main/java/io/prestosql/plugin/hive/HiveSessionProperties.java

```diff
@@ -363,7 +367,22 @@ public HiveSessionProperties(
                         PROJECTION_PUSHDOWN_ENABLED,
                         "Projection push down enabled for hive",
                         hiveConfig.isProjectionPushdownEnabled(),
-                        false));
+                        false),
+                new PropertyMetadata<>(
+                        HIVE_CACHE_ENABLED,
+                        "Enable Hive caching",
+                        BOOLEAN,
+                        Boolean.class,
+                        rubixEnabledConfig.isCacheEnabled(),
+                        false,
+                        value -> {
+                            boolean booleanValue = (boolean) value;
+                            if (booleanValue && !rubixEnabledConfig.isCacheEnabled()) {
+                                throw new PrestoException(INVALID_SESSION_PROPERTY, "Cache can only be disabled via session property");
+                            }
+                            return booleanValue;
+                        },
+                        value -> value));
```

Comment:
`Function::identity`?

Reply:
There is casting involved, so it doesn't work